test/test_creating_dataset.py

```diff
@@ -26,89 +18,45 @@
 
 
 # is_path_correct
-def test_check_paths_wrong_path():
-    assert not is_path_correct("pathfinder_2e_data/pathfinder-bestiar.db")
+def test_is_path_correct():
+    with pytest.raises(ValueError):
+        # case: path does not exist
+        is_path_correct("pathfinder_2e_data/pathfinder-bestiar.db")
 
+    with pytest.raises(ValueError):
+        # case: wrong file type
+        is_path_correct("requirements.txt")
 
-def test_check_paths_wrong_file_type():
-    assert not is_path_correct("requirements.txt")
-
-
-def test_check_right_path():
+    # case: correct path
     assert is_path_correct(DATASET_PATHS[0])
 
 
-# get_subcolumn
-def test_get_subcolumn():
-    system = unpack_column(BESTIARY, "system")
-    pd.testing.assert_frame_equal(system, get_subcolumn(BESTIARY, "system"))
-
-    abilities = unpack_column(system, "abilities")
-    pd.testing.assert_frame_equal(
-        abilities, get_subcolumn(BESTIARY, "system/abilities")
-    )
-
-
-def test_wrong_key_get_subcolumn():
-    with pytest.raises(KeyError):
-        get_subcolumn(BESTIARY, "abilities/system")
-
-
-# move_values_level_up
-def test_move_values_level_up():
-    level_up_function = move_values_level_up("mod")
-    abilities = get_subcolumn(BESTIARY, "system/abilities")
-    test_abilities = deepcopy(abilities)
-
-    level_up_abilities = level_up_function(abilities)
-    for col in test_abilities.columns:
-        for i, row in test_abilities.iterrows():
-            row[col] = row[col].get("mod")
-
-    test_abilities = test_abilities.apply(pd.to_numeric)
-
-    pd.testing.assert_frame_equal(level_up_abilities, test_abilities)
-
-
-# load_subcolumn_as_value
-def test_load_subcolumn_as_value():
-    ac = get_subcolumn(BESTIARY, "system/attributes/ac")
-    ac_test = deepcopy(ac)
-
-    load_ac = load_subcolumn_as_value("ac")
-    ac = load_ac(ac)
-    ac_test = pd.DataFrame(data=ac_test.value)
-    ac_test.columns = ["ac"]
-
-    pd.testing.assert_frame_equal(ac, ac_test)
-
-
-# _create_df_with_basic_values
-def test_create_df_with_basic_values():
-    basic_bestiary = _create_df_with_basic_values(BESTIARY)
-
-    books = load_subcolumn_as_value("book")(
-        get_subcolumn(BESTIARY, "system/details/source")
-    )
-    levels = load_subcolumn_as_value("level")(
-        get_subcolumn(BESTIARY, "system/details/level")
-    )
+# load_and_preprocess_data
+def test_load_and_preprocess_data():
+    test_dataframe = pd.read_json("../output/bestiary_system_basic.json")
 
-    pd.testing.assert_frame_equal(pd.DataFrame(data=basic_bestiary["book"]), books)
+    column_paths = {
+        "book": "system/details/source",
+        "focus": "system/resources/focus",
+        # saves: fortitude, reflex, will
```

Comment:
W sumie czy potrzebujemy tego komentarza?

Reply:
Został wprowadzony tylko po to, żeby było wiadomo dlaczego `test_dataframe` będzie zawierało akurat te a nie inne cechy. Dla mnie jest to raczej oczywiste jakie są składowe `saves`, ale nie wiem czy dla mnie z przyszłości będzie